pkg/autoscaler/autoscaler/tikv_autoscaler.go

```diff
@@ -111,6 +118,35 @@ func calculateTiKVMetrics(tac *v1alpha1.TidbClusterAutoScaler, tc *v1alpha1.Tidb
 	return nil
 }
 
+func calculateTiKVExternalService(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, sts *appsv1.StatefulSet, kubecli kubernetes.Interface) error {
+	targetReplicas, err := query.ExternalService(tc, v1alpha1.TiKVMemberType, tac.Spec.TiKV.ExternalEndpoint, kubecli)
+	if err != nil {
+		klog.Errorf("tac[%s/%s] 's query to the external endpoint got error: %v", tac.Namespace, tac.Name, err)
+		return err
+	}
+	targetReplicas = limitTargetReplicas(targetReplicas, tac, v1alpha1.TiKVMemberType)
+	if targetReplicas == tc.Spec.TiKV.Replicas {
+		return nil
+	}
+	currentReplicas := tc.Spec.TiKV.Replicas
+	intervalSeconds := tac.Spec.TiKV.ScaleInIntervalSeconds
+	if targetReplicas > currentReplicas {
+		intervalSeconds = tac.Spec.TiKV.ScaleOutIntervalSeconds
+	}
+	ableToScale, err := checkTiKVAutoScalingInterval(tac, *intervalSeconds)
+	if err != nil {
+		return err
+	}
+	if !ableToScale {
+		return nil
+	}
+	err = updateTacIfTiKVScale(tc, tac, targetReplicas)
+	if err != nil {
+		return err
+	}
+	return addAnnotationMarkIfScaleOutDueToCPUMetrics(tc, currentReplicas, targetReplicas, sts)
```

Comment:
do we need to mark the scale-out pod in external service case? This would add label `specialUse:Hot` store label for it.

Reply:
the rest LGTM.

Reply:
Yes, I think it‘s required in external service case,  the scale-out pod should only balance hot region too.